step_5/backend/main.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	opentracingLog "github.com/opentracing/opentracing-go/log"
+	"log"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/inovex/godays-demo/pkg"
+)
+
+var port int
+
+func init() {
+	flag.IntVar(&port, "port", 8080, "port to listen on")
+}
+
+func toastsHandler(w http.ResponseWriter, r *http.Request) {
+	context, err := opentracing.GlobalTracer().Extract(opentracing.HTTPHeaders, opentracing.HTTPHeadersCarrier(r.Header))
+	if err != nil {
+		log.Printf("Could not extract SpanContext from request %s", err)
+	}
+	span := opentracing.GlobalTracer().StartSpan("/toasts", opentracing.ChildOf(context))
+	defer span.Finish()
+
+	// Set tags for filtering
+	ext.SpanKindRPCServer.Set(span)
+	ext.HTTPUrl.Set(span, r.URL.String())
+	ext.HTTPMethod.Set(span, r.Method)
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if rand.Int()%4 != 0 {
```

Comment:
```suggestion
        // return the toast slice 3 out of 4 calls, else return a HTTP 500
	if rand.Int()%4 != 0 {
```

Please add a comment on the intent of this rand calculation